### PR TITLE
[MCC-425214] implement httpmodule

### PR DESCRIPTION
**Pull request by kenyamat**
@mdsol/team-51 @cabbott please review and merge.

### CODE REVIEW

File: src/Medidata.ZipkinTracer.HttpModule/ZipkinRequestContextModule.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        {
+            context.BeginRequest += BeginRequest;
+            context.EndRequest += EndRequest;
+        }
+
+        public void BeginRequest(object sender, EventArgs e)
+        {
+            var application = (HttpApplication)sender;
+            var request = _contextHelper.GetRequest(application.Context);
+            var response = _contextHelper.GetResponse(application.Context);
```
Comment (bvillanueva-mdsol) [inline src/Medidata.ZipkinTracer.HttpModule/ZipkinRequestContextModule.cs]: we can remove response line. not being used

File: src/Medidata.ZipkinTracer.HttpModule/HttpContextExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+            TraceManager.SamplingRate = config.ZipkinSampleRate;
+            TraceManager.RegisterTracer(new zipkin4net.Tracers.Zipkin.ZipkinTracer(
+                new HttpZipkinSender(config.ZipkinBaseUri.AbsoluteUri, "application/json"),
+                new JSONSpanSerializer()
+            ));
+            TraceManager.Start(logger);
+        }
+
+        public static bool IsZipkinHttpModuleEnabled(this HttpContext context)
```
Comment (bvillanueva-mdsol) [inline src/Medidata.ZipkinTracer.HttpModule/HttpContextExtensions.cs]: we can mark this as private or internal if needed for unit test

File: src/Medidata.ZipkinTracer.HttpModule/ZipkinConfig.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Configuration;
+
+namespace Medidata.ZipkinTracer.HttpModule
+{
+    public class ZipkinConfig : IZipkinConfig
```
Comment (bvillanueva-mdsol) [inline src/Medidata.ZipkinTracer.HttpModule/ZipkinConfig.cs:7]: @kenyamat I suggest to have this implemented at Gambit or the client application side. Example is Get method can be getting settings key from environment variables

**Comment by kenyamat** [comment]
I will move this class to gambit repo.

**Pull request closed without merging**